app/models/offer.rb

```diff
@@ -131,8 +124,13 @@ def oms_params_match?
   end
 
   def same_order_type_as_in_service
-    unless order_type == service.order_type
+    if new_record? && service.offers.published.empty? && order_type != service.order_type
       errors.add(:order_type, "must be the same as in the service: #{service.order_type}")
```

Comment:
You have code duplication and muliple nested if statements. What do you think about writing something like this?
```
  def same_order_type_as_in_service
    other_services = service.offers.published.reject { |o| o&.id == id }
    other_check = other_services.none? { |o| o.order_type == service&.order_type }
    if (other_services.empty? || other_check ) && order_type != service.order_type
      errors.add(:order_type, "must be the same as in the service: #{service.order_type}")
    end
  end
```